app/angular/src/client/preview/angular-beta/AbstractRenderer.ts

```diff
@@ -0,0 +1,193 @@
+/* eslint-disable no-undef */
+import { enableProdMode, NgModule, PlatformRef } from '@angular/core';
+import { platformBrowserDynamic } from '@angular/platform-browser-dynamic';
+
+import { BehaviorSubject, Subject } from 'rxjs';
+import { ICollection, StoryFnAngularReturnType } from '../types';
+import { Parameters } from '../types-6-0';
+import { createStorybookModule, getStorybookModuleMetadata } from './StorybookModule';
+
+type StoryRenderInfo = {
+  storyFnAngular: StoryFnAngularReturnType;
+  moduleMetadataSnapshot: string;
+};
+
+// platform must be init only if render is called at least once
+let platformRef: PlatformRef;
+function getPlatform(newPlatform?: boolean): PlatformRef {
+  if (!platformRef || newPlatform) {
+    platformRef = platformBrowserDynamic();
+  }
+  return platformRef;
+}
+
+export abstract class AbstractRenderer {
+  /**
+   * Wait and destroy the platform
+   */
+  protected static resetPlatformBrowserDynamic() {
+    return new Promise<void>((resolve) => {
+      resolve();
```

Comment:
Why do you need to resolve it twice? And why at the start of the promise?

Reply:
🐛